### DISCUSSION TITLE
Generalized Maxwell Model - convergence issues and negative Jacobians

### DISCUSSION THREAD

**Discussion by s-dunnim** [idaholab/moose#17092] [Q&A Modules: Solid mechanics] [answered]
Hi,

I'm trying to model a compression moulding process, using a GeneralizedMaxwellModel. I'm using a generated mesh for the charge to be moulded, and another pre-made mesh for the inside edge of the mould (see picture). The bottom of the mould (-y) is modelled by a 0 displacement BC, and the top of the mould (+y), which moves downwards to compress the charge, is modelled with a variable disp BC. There is a 0 disp BC along the left face of the charge (-x), to stop it from separating from the inside of the mould (this would be due to friction from the top and bottom of the mould in real life), although the corner is free, allowing it to fill the rounded corner of the mould. The charge has been halved (in the xy plane), and a 0 disp BC applied to the front face (+z), using symmetry to reduce the calculation required. Frictionless contact is enforced between the inner mould surface and the outer charge surfaces.

<img width="1116" alt="Charge and Mould mesh" src="https://user-images.githubusercontent.com/78145987/108738363-701e2780-752b-11eb-9811-220196dacbf0.png">

Unfortunately, the model fails about halfway through, producing multiple negative Jacobian errors and failing to converge. I've tried adaptive time stepping, mesh adaptivity, different contact formulations, turning off contact completely, and different sizes of geometry, none of which fixed the problem (although different configurations did result in the failure occurring at different times).

Does anyone know what the problem could be, and what I can do to fix it? Ideally I'd like to get the charge squashing out to fill the mould shape. I've attached the input file, as well as the mould mesh and the .csv used for the mould top height.

Thanks

[CompressionMoulding.i.txt](https://github.com/idaholab/moose/files/6023449/CompressionMoulding.i.txt)
[MouldMesh.ucd.txt](https://github.com/idaholab/moose/files/6023451/MouldMesh.ucd.txt)
[MouldTopDisp.csv.txt](https://github.com/idaholab/moose/files/6023464/MouldTopDisp.csv.txt)

**Comment by GiudGiud** [comment, marked as answer]
Hi
Did you check the mesh for bad displaced/inverted elements before the solver fails?
@aeslaughter who can I tag to help with this?

Guillaume

**Reply by s-dunnim** [reply]
Hi there, thanks for the reply. Some of the elements were pretty highly deformed and displaced - in the end I came to the conclusion the strain and deformation levels were perhaps more than TensorMechanics is really made for, and I'm investigating other methods to model this in MOOSE instead.

**Reply by GiudGiud** [reply]
Ok thanks a lot for letting us know.
